collect_app/src/main/java/org/odk/collect/android/upload/InstanceUploader.java

```diff
@@ -49,55 +42,26 @@
      * Returns a list of Instance objects corresponding to the database IDs passed in.
      */
     public List<Instance> getInstancesFromIds(Long... instanceDatabaseIds) {
-        List<Instance> instancesToUpload = new ArrayList<>();
-        InstancesDao dao = new InstancesDao();
-
-        // Split the queries to avoid exceeding SQLITE_MAX_VARIABLE_NUMBER
-        int counter = 0;
-        while (counter * ApplicationConstants.SQLITE_MAX_VARIABLE_NUMBER < instanceDatabaseIds.length) {
-            int low = counter * ApplicationConstants.SQLITE_MAX_VARIABLE_NUMBER;
-            int high = (counter + 1) * ApplicationConstants.SQLITE_MAX_VARIABLE_NUMBER;
-            if (high > instanceDatabaseIds.length) {
-                high = instanceDatabaseIds.length;
-            }
-
-            StringBuilder selectionBuf = new StringBuilder(InstanceColumns._ID + " IN (");
-            String[] selectionArgs = new String[high - low];
-            for (int i = 0; i < (high - low); i++) {
-                if (i > 0) {
-                    selectionBuf.append(',');
-                }
-                selectionBuf.append('?');
-                selectionArgs[i] = instanceDatabaseIds[i + low].toString();
-            }
-
-            selectionBuf.append(')');
-            String selection = selectionBuf.toString();
+        List<Instance> instances = new ArrayList<>();
 
-            Cursor c = dao.getInstancesCursor(selection, selectionArgs);
-            instancesToUpload.addAll(dao.getInstancesFromCursor(c));
-
-            counter++;
+        for (Long id : instanceDatabaseIds) {
+            instances.add(new DatabaseInstancesRepository().get(id));
```

Comment:
Again this could be more efficient but not sure it's a big deal given the instance count will usually be low.